BitSight Solution initial submission for certification.

jayeshprajapaticrest opened the pull request:
Required items, please complete
   
   Change(s):
   - Added BitSight Solution for certification.

   Reason for Change(s):
   - New product

   Version Updated:
   - Initial version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

jayeshprajapaticrest commented on Solutions/BitSight/Playbooks/GetWFHData/README.md:
```diff
@@ -0,0 +1,62 @@
+### Deployment instructions<a name="Deployment-instructions"></a>
+
+1. To deploy the Playbook, click the Deploy to Azure button. This will launch the ARM Template deployment wizard.
+2. Fill in the required paramteres:
```
@v-sabiraj I have fixed the Typo error and committed it to this PR.
Let me know if any other changes require.

v-sabiraj commented:
@jayeshprajapaticrest, can you please remove the keys in analytic rules which are empty, thanks.

jayeshprajapaticrest commented:
> @jayeshprajapaticrest, can you please remove the keys in analytic rules which are empty, thanks.

@v-sabiraj I have removed the empty keys from JSON and updated solution accordingly.
Let me know if anything is required.

One request can you please take this PR on priority to as we need this solution publicly as possible we can?

manishkumar1991 commented on Solutions/BitSight/Playbooks/GetWFHData/azuredeploy.json:
```diff
@@ -0,0 +1,854 @@
+                                "connectionId":  "[resourceId('Microsoft.Web/connections', variables('KeyvaultConnectionName'))]",
+                                "connectionName":  "[variables('KeyvaultConnectionName')]",
+                                "id":  "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location, '/managedApis/Keyvault')]"
+                            }
```
Key vault connection should be managed service identity.

Kindly refer to Below playbook for ref 
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ThreatXCloud/Playbooks/ThreatXPlaybooks/ThreatX-BlockIP-URL/azuredeploy.json

jayeshprajapaticrest replied:
@manishkumar1991 
We have added managed service identity for Key vault for playbook.

Question:
Just to know can you please let me know what impact or extra steps is required for end user who is going to use/deploy this playbook once it publicly available?

v-sabiraj commented:
@jayeshprajapaticrest, can you please fix the validation errors, thanks.

jayeshprajapaticrest commented:
> @jayeshprajapaticrest, can you please fix the validation errors, thanks.

@v-sabiraj I have fixed the validation errors. Can you further proceed with the PR merge?
Let me know if any other changes required.

v-sabiraj commented:
@devikamehra, can you please check on the Analytic Rules and approve. Thanks.

v-vdixit commented:
Hi @devikamehra could you please check analytic rules, thanks!

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_when_there_is_a_drop_of_10%_or_more.yaml:
```diff
@@ -0,0 +1,35 @@
+id: d8844f11-3a36-4b97-9062-1e6d57c00e37
+name: Alert when there is a drop of 10% or more
```
@devikamehra 
We have followed the contribution guide and fixed the changes of filename, analytics rule name and description.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_when_there_is_a_drop_of_10%_or_more.yaml:
```diff
@@ -0,0 +1,35 @@
+triggerOperator: GreaterThan
+triggerThreshold: 0
+query: |
+    let timeframe = 24h;
```
@devikamehra 
We have fixed the changes as suggested above.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_when_there_is_a_drop_of_10%_or_more.yaml:
```diff
@@ -0,0 +1,35 @@
+name: Alert when there is a drop of 10% or more
+description: |
+  'Creates an incident when there is a drop of 10% or more in BitSight company ratings.'
+severity: High
```
@devikamehra 
We have added Tactic and Techniques as per match with the Analytics rule.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_when_there_is_a_new_alert_from_BitSight.yaml:
```diff
@@ -0,0 +1,40 @@
+    let timeframe = 24h;
+    BitSightAlerts
+    | where ingestion_time() > ago(timeframe)
+    | extend Severity = case( Severity contains "INFORMATIONAL", 0,
```
@devikamehra 
We have updated the Analytics rule with the mapping of the Severity value as per uniformity.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_when_there_is_a_new_alert_from_BitSight.yaml:
```diff
@@ -0,0 +1,40 @@
+                              Severity contains "WARN" or Severity contains "DECREASE", 2,
+                              Severity contains "CRITICAL", 3,
+                              0)
+    | extend CompanyURL = strcat("https://service.bitsighttech.com/app/spm",CompanyURL)
```
@devikamehra 
Yes, It will be the same across all the customer.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_whenever_there_is_a_Diligence_finding_in_BitSight.yaml:
```diff
@@ -0,0 +1,35 @@
+id: 161ed3ac-b242-4b13-8c6b-58716e5e9972
+name: Alert whenever there is a Diligence finding in BitSight
```
@devikamehra 
Fixed the changes please check the latest commit.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_whenever_there_is_a_compromised_systems_finding_in_BitSight.yaml:
```diff
@@ -0,0 +1,35 @@
+id: d68b758a-b117-4cb8-8e1d-dcab5a4a2f21
+name: Alert whenever there is a compromised systems finding in BitSight
```
@devikamehra 
Fixed the grammatical mistake.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_whenever_there_is_a_compromised_systems_finding_in_BitSight.yaml:
```diff
@@ -0,0 +1,35 @@
+    let timeframe = 24h;
+    BitSightFindingsData
+    | where ingestion_time() > ago(timeframe)
+    | where RiskCategory == "Compromised Systems"
```
@devikamehra 
Sample data to test the above analytics rule is provided.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alert_whenever_there_is_a_drop_in_the_headline_rating.yaml:
```diff
@@ -0,0 +1,34 @@
+id: b11fdc35-6368-4cc0-8128-52cd2e2cdda0
+name: Alert whenever there is a drop in the headline rating
```
@devikamehra 
We understood your question but it would be required as client needed this for their usage.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alerts_for_a_new_breach.yaml:
```diff
@@ -0,0 +1,34 @@
+alertDetailsOverride:
+    alertDisplayNameFormat: 'BitSight: Alert for new breach in {{Companyname}}.'
+    alertDescriptionFormat: 'Alert is generated on {{DateCreated}} at BitSight.\n\nGUID: {{GUID}}\nPreview URL: {{PreviwURL}}'
+    alertSeverityColumnName: Severity
```
@devikamehra 
We have updated the Analytics rule with the mapping of the Severity value as per uniformity.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alerts_when_there_are_new_Botnet_Infections.yaml:
```diff
@@ -0,0 +1,35 @@
+    let timeframe = 24h;
+    BitSightFindingsData
+    | where ingestion_time() > ago(timeframe)
+    | where RiskVector == "botnet_infections"
```
@devikamehra 
Sample data to test the above analytics rule is provided.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/Alerts_when_there_is_a_new_finding_for_Patching_Cadence.yaml:
```diff
@@ -0,0 +1,35 @@
+    let timeframe = 24h;
+    BitSightFindingsData
+    | where ingestion_time() > ago(timeframe)
+    | where RiskVector == "patching_cadence"
```
@devikamehra 
Sample data to test the above analytics rule is provided.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSight_Get_WFH_Data.yaml:
```diff
@@ -0,0 +1,33 @@
+id: 54e77434-2184-435f-bc87-52b832ee451f
+name: BitSight Get WFH Data
```
@devikamehra 
We have updated the file name, analytics rule name and the description which does not contains the abbreviations.

jayeshprajapaticrest commented:
> There are multiple issues which are common to all the queries
> 
> 1. Improve the Name and Description of rules.
> 2. queryPeriod and lookback time do not match
> 3. Tactic and techniques are missing
> 4. All queries are HIGH in severity. Please divide them amongst different severity levels.
> 5. There are no Entity or custom details which are getting matched. Try matching atleast one entity wherever possible.

@devikamehra 
We have fixed all the above common changes to all the Analytics rule.
So please confirm and let me know if any other changes required from our end.
Thanks

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSight_Get_WFH_Data.yaml:
```diff
@@ -0,0 +1,33 @@
+triggerThreshold: 0
+query: |
+    ASimNetworkSession
+    | where EventSchema == "NetworkSession"
```
@devikamehra 
We have specify that the user must needs to setup ASIM parser for already existing vendors in sentinel in the Readme file. We are not going to ship the ASIM parser for this as its not required the use case is as below.

Any user who is going to use this integration and they have different source of data in the sentinel other than the BitSight data and they want to use this Work From Home functionality then its their responsibility to create ASIM parser of NetworkSession schema and use this analytics rule which pass the other sources network session's src ip address to the playbook which would be executed using this analytics rule triggering. And then call the BitSight API with this mapped ip address and if data would be found related to this ip in the bitsight then it will store in the BitSight Work From Home custom table and same populated in the WFM dashboard of workbook.

devikamehra replied:
@jayeshprajapaticrest 
I think it is not how OOTB content works in Sentinel. Either we provide the ASIM parser out of the box and then give this query or we do not include this query at all.
Also, anything related to Network Session is already covered in Network Session Essential solution. We do not need to keep this in product specific solutions.
Please remove this use case for now.

devikamehra replied:
Plus, the query is not doing much here. It is creating Alert for all the Network Session information. This will be extremely noisy and irrelevant for any SOC Analyst.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightDropInCompanyRatings.yaml:
```diff
@@ -0,0 +1,42 @@
+triggerOperator: GreaterThan
+triggerThreshold: 0
+tactics:
+  - Command And Control
```
@v-sabiraj Updated as above. Thanks

v-vdixit replied:
Hi @jayeshprajapaticrest we are not able to see the updated tactics as 'CommandAndControl', please check why the changes are not reflecting, thanks!

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightDropInHeadlineRating.yaml:
```diff
@@ -0,0 +1,41 @@
+triggerOperator: GreaterThan
+triggerThreshold: 0
+tactics:
+  - Command And Control
```
@v-sabiraj Updated as above. Thanks

v-vdixit replied:
Hi @jayeshprajapaticrest please check why the changes are not reflecting, thanks!

v-sabiraj commented:
@jayeshprajapaticrest, will check on other things, thanks.

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightBadOpenPortsDetected.yaml:
```diff
@@ -0,0 +1,47 @@
+incidentConfiguration:
+    createIncident: true
+    groupingConfiguration:
+      enabled: false
```
groupingConfiguration is set to false here. If not required, please remove from the YAML.

jayeshprajapaticrest replied:
@devikamehra We have removed the above grouping block as it was set to true and there is no meaning of that. Thanks

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightDropInHeadlineRating.yaml:
```diff
@@ -0,0 +1,41 @@
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: CompanyName
```
Please change the mapping to custom details

devikamehra replied:
Not done

v-atulyadav commented:
Hi @jayeshprajapaticrest, please check comments above and act accordingly. Thanks

jayeshprajapaticrest commented:
> Hi @jayeshprajapaticrest, please check comments above and act accordingly. Thanks

@v-atulyadav @devikamehra 
Will check above and fixing it. Thanks

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightBadOpenPortsDetected.yaml:
```diff
@@ -0,0 +1,47 @@
+  - entityType: Malware
+    fieldMappings:
+      - identifier: Name
+        columnName: RiskVector
```
@devikamehra Done with the changes as suggested above.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightBadOpenPortsDetected.yaml:
```diff
@@ -0,0 +1,47 @@
+    fieldMappings:
+      - identifier: Name
+        columnName: RiskVector
+version: 1.0.1
```
@devikamehra Updated version as 1.0.0 as suggested above.

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightDropInCompanyRatings.yaml:
```diff
@@ -0,0 +1,42 @@
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: CompanyName
```
@devikamehra Done. Thanks

devikamehra replied:
This change has not been done.

jayeshprajapaticrest commented:
@devikamehra 
Regarding the WorFromHome Analytics rule relate question.
We have understood your concern, but the actual use case of this analytic rule is as below, 
1) We are using this analytic rule as a sample rule for the users.
2) We have mapped "Src" field of the ASIM network session parser with customDetails field of analytic rule with the name "ip_address".
3) The user will add one automation rule after deployment of this analytic rule to run a workflow and that workflow will use our mapped field "ip_address" to fetch the data of BitSight.
4) Also workflow will store that fetched BitSight data into log analytic workspace and will displaying the data in workbook.

Also by default this rule will be disabled and if user want the data of network session from every configured sources which only has High EventSeverity then user can modify the query of this analytic rule and can access our connected workflow as explained above.

Let me know if you have any questions related this. 
Thanks.

v-atulyadav commented:
Hi @devikamehra, suggested changes fixed by @jayeshprajapaticrest please have a look. Thanks,

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightBadOpenPortsDetected.yaml:
```diff
@@ -0,0 +1,47 @@
+                              toreal(Severity) <= 6.9 and toreal(Severity) >= 4.0, "Low",
+                              toreal(Severity) <= 8.9 and toreal(Severity) >= 7.0, "Medium",
+                              toreal(Severity) <= 10.0 and toreal(Severity) >= 9.0, "High",
+                              "Informational")
```
| extend Severity = case(toreal(Severity) <= 6.9 and toreal(Severity) >= 4.0, "Low",
                                          toreal(Severity) <= 8.9 and toreal(Severity) >= 7.0, "Medium",
                                          toreal(Severity) <= 10.0 and toreal(Severity) >= 9.0, "High",
                                          "Informational")

jayeshprajapaticrest replied:
@devikamehra 
Changes done as suggested above.
Thanks

jayeshprajapaticrest commented on Solutions/BitSight/Analytic Rules/BitSightBadOpenPortsDetected.yaml:
```diff
@@ -0,0 +1,47 @@
+      lookbackDuration: 24h
+      matchingMethod: AllEntities
+eventGroupingSettings:
+    aggregationKind: AlertPerResult
```
@devikamehra 
Yes exactly, we want to create each alerts for every detected bad opened port.
If any port is open then the new data for that is created at BitSight and our data connector will ingest it in sentinel and as a use case we need to generate incident for each open port.

devikamehra commented on Solutions/BitSight/Parsers/BitSightAlerts.txt:
```diff
@@ -0,0 +1,42 @@
+let Alerts_view  = view () { 
+    Alerts_data_CL
+    | extend 
+                EventVendor ="AlertBitSight",
```
Vendor should be BitSight and Product should be Alert.
This is applicable to all the Parsers.

jayeshprajapaticrest replied:
@devikamehra Changes done as above.

devikamehra commented on Solutions/BitSight/Parsers/BitSightBreaches.txt:
```diff
@@ -0,0 +1,44 @@
+    | extend 
+                EventVendor ="BreachesBitSight",
+                EventProduct ="BreachesBitSight",
+                GUID = column_ifexists('guid_d', ''),
```
It can be guid_g as well. Consider both in this case.

jayeshprajapaticrest replied:
@devikamehra Changes done as above.

devikamehra commented on Solutions/BitSight/Parsers/BitSightCompanyDetails.txt:
```diff
@@ -0,0 +1,114 @@
+                EventVendor ="CompanyDetailsBitSight",
+                EventProduct ="CompanyDetailsBitSight",
+                PrimaryCompanyGUID = column_ifexists('primary_company_guid_g', ''),
+                PrimaryCompanyName = column_ifexists('primary_company_name_s', ''),
```
The sample data contained company_name only and thus will not get mapped.

jayeshprajapaticrest replied:
@devikamehra I have updated correct sample data.

devikamehra commented on Solutions/BitSight/Parsers/BitSightCompanyDetails.txt:
```diff
@@ -0,0 +1,114 @@
+                ComplianceClaimCertifications = column_ifexists('compliance_claim_certifications_s', ''),
+                ComplianceClaimTrustPage = column_ifexists('compliance_claim_trust_page_s', ''),
+                type = column_ifexists('type_s', ''),
+                Type = column_ifexists('Type', '')
```
Many fields are not present in the sample data and thus cannot be tested.

jayeshprajapaticrest replied:
@devikamehra I have updated correct sample data.

devikamehra replied:
Thank you for updating the sample data.
Type is a reserved field and thus will always be there in the custom field. We do not need column_ifexists.

jayeshprajapaticrest commented:
@devikamehra @v-atulyadav 
A gentle request to please provide a final list of changes covering everything related to analytic rule in one go only so I can resolve them at once and we can hasten up the merging process.

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightCompromisedSystemsDetected.yaml:
```diff
@@ -0,0 +1,44 @@
+    let timeframe = 24h;
+    BitSightFindingsData
+    | where ingestion_time() > ago(timeframe)
+    | where RiskCategory == "Compromised Systems"
```
According to the sample data, RiskCategory == "Compromised Systems" will cover "botnet_infections". This will be a duplicate with BigSightBotnetInfectionDetection.
Either remove the BotnetInfectionDetection or make this detection more specific.

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightDropInHeadlineRating.yaml:
```diff
@@ -0,0 +1,36 @@
+  - entityType: Sentinel entities
+    fieldMappings:
+      - identifier: Entities
+        columnName: AlertId
```
AlertId is not project and will not get populated.
Same is applicable to all the queries where we are referring to this field.

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightGetWorkFromData.yaml:
```diff
@@ -0,0 +1,35 @@
+id: 54e77434-2184-435f-bc87-52b832ee451f
```
Please remove this query as discussed before.

jayeshprajapaticrest replied:
@devikamehra @v-atulyadav 
After internal discussion we have decided to remove the Work From Home use case completely.
And as a action we have removed the Analytics Rule, Playbook and Workbook related to Work From Home.

Thanks.

devikamehra commented on Solutions/BitSight/Analytic Rules/BitSightPatchingCadenceDetected.yaml:
```diff
@@ -0,0 +1,44 @@
+    let timeframe = 24h;
+    BitSightFindingsData
+    | where ingestion_time() > ago(timeframe)
+    | where RiskVector == "patching_cadence"
```
This query is also a duplicate with RiskCategory == "Diligence".

jayeshprajapaticrest replied:
@devikamehra @v-atulyadav  We have removed the PatchingCadence analytics rule.
After internal discussion we have also removed the BitSightBadOpenPortsDetected analytics rule which would not be required as usecase from now.
Thanks.

devikamehra commented:
@v-atulyadav Please get the other components reviewed. I have reviewed the analytical query.

v-sabiraj commented:
@jayeshprajapaticrest, can you fix the "Command And Control" to "CommandAndControl", also please check on the other comments.

jayeshprajapaticrest commented:
> @jayeshprajapaticrest, can you fix the "Command And Control" to "CommandAndControl", also please check on the other comments.

@v-sabiraj We have fixed the changes suggested above. Thanks

devikamehra reviewed (dismissed):
Only analytical rules are validated. Please feel free to merge in case everything else is good.

v-prasadboke commented:
Hello @v-sabiraj please look into this

jayeshprajapaticrest commented:
@v-prasadboke @v-sabiraj 
Is there any update on this PR approval? As we need this solution available to the azure marketplace as early as possible.

v-atulyadav reviewed (commented):
Hi @jayeshprajapaticrest, please check below comments.

1. Metadata should not be in this path, you should update metadata from "Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json". Remove the metadata folder from this solution

2. The images should be in the Preview folder, which can be found at Workbooks/Images/Preview. Below screenshot is for your reference
<img width="244" alt="image" src="https://user-images.githubusercontent.com/104008048/233081459-8abf8ac3-7687-4ef6-a54d-fa0be03a9835.png">

jayeshprajapaticrest commented:
> Hi @jayeshprajapaticrest, please check below comments.
> 
> 1. Metadata should not be in this path, you should update metadata from "Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json". Remove the metadata folder from this solution
> 2. The images should be in the Preview folder, which can be found at Workbooks/Images/Preview. Below screenshot is for your reference
> 
> <img alt="image" width="244" src="https://user-images.githubusercontent.com/104008048/233081459-8abf8ac3-7687-4ef6-a54d-fa0be03a9835.png">

@v-atulyadav I have completed the changes as suggested above. Thanks
Can you please proceed with the PR approval related process?

v-rbajaj commented:
Hi @jayeshprajapaticrest, can you please fix the validations?

jayeshprajapaticrest commented:
> Hi @jayeshprajapaticrest, can you please fix the validations?

@v-rbajaj Done with the fixing of validations.
Can you please proceed it?

jayeshprajapaticrest commented:
@v-atulyadav We have fixed the validation issues. So can you please proceed with the merging the PR as early as possible from your end so we can get our solution publicly available.

v-prasadboke commented:
Hello @jayeshprajapaticrest looking into this

jayeshprajapaticrest commented:
@v-prasadboke @v-atulyadav 
We are done with the all the changes and the validation fixings.
So can you please expediate the process so we can get the solution available in the azure marketplace as early as we can.

v-atulyadav commented:
Hi @jayeshprajapaticrest, thanks for your response we will expediate the process.

v-prasadboke commented:
hello @jayeshprajapaticrest the tables dont exist by their name after ingesting the sample data,
<img width="788" alt="image" src="https://user-images.githubusercontent.com/117061676/234813499-656c0644-f76a-4b4d-be2e-5ef93236de63.png">

Had to rewrite it as 
BitSightGraphData_CL, But still it doesnt retrieve the data
<img width="775" alt="image" src="https://user-images.githubusercontent.com/117061676/234814136-16ea355e-67e6-4814-b4a3-347732c9789b.png">

v-prasadboke commented:
same is the case with BitSightFindingData

jayeshprajapaticrest commented:
@v-prasadboke 
We have provided the sample data in the path below.
[Azure-Sentinel](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution)/[Sample Data](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution/Sample%20Data)/[Custom](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution/Sample%20Data/Custom)/BitSight/
You can create the table name as Graph_data_CL from the sample data provided at above path. And to use can configure the custom parser with the name as BitSightGraphData which refers the custom table name as Graph_data_CL which we created above from the provided sample data.

And same for the BitSightFindingData as well.

Note: We have created the sample data files with the custom table names which further referenced in the Custom Parsers as well.
Let me know if we need to update the sample data file name or the parser or may be the table name referenced in the custom parser.
Thanks.

Cc: @v-atulyadav

jayeshprajapaticrest commented:
@v-prasadboke Let me know if the above works or if we need to do any changes so we will do it and expediate this process.
Thanks.
Cc: @v-atulyadav

v-prasadboke commented:
> @v-prasadboke We have provided the sample data in the path below. [Azure-Sentinel](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution)/[Sample Data](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution/Sample%20Data)/[Custom](https://github.com/jayeshprajapaticrest/Azure-Sentinel/tree/BitSight-Solution/Sample%20Data/Custom)/BitSight/ You can create the table name as Graph_data_CL from the sample data provided at above path. And to use you can configure the custom parser with the name as BitSightGraphData which refers the custom table name as Graph_data_CL which is created from above provided sample data.
> 
> And same for the BitSightFindingData as well.
> 
> Note: We have created the sample data files with the custom table names which further referenced in the Custom Parsers as well. Let me know if we need to update the sample data file name or the parser or may be the table name referenced in the custom parser. Thanks.
> 
> Cc: @v-atulyadav

Ok @jayeshprajapaticrest let me go through this

v-prasadboke commented:
Ok @jayeshprajapaticrest going through this

jayeshprajapaticrest commented:
> Ok @jayeshprajapaticrest going through this

@v-prasadboke Is there any update on this?

v-prasadboke commented:
Sorry @jayeshprapajaticrest for the late reply, still working on this

v-prasadboke commented:
Hello @jayeshprajapaticrest sorry for taking to much time, I was facing problem with sample data ingestion.
But now the sample data ingestion is working fine.
Wont take much time to test the content of the solution

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest sorry for taking to much time, I was facing problem with sample data ingestion. But now the sample data ingestion is working fine. Wont take much time to test the content of the solution

@v-prasadboke 
OK Understood. Thanks

v-prasadboke commented:
Hello @jayeshprajapaticrest Diligence_statistics_CL,
Findings_data_CL, Breaches_data_CL  data ingetion for this 3 sample data were unsuccessfull. Due to which with some of the analytic rule and parser I'm facing some issues.

While testing parser query it was failing, but if instructions removed from the query it was running good, is this the same case with you or its with me only.

Thank you, Prasad`

jayeshprajapaticrest commented:
@v-prasadboke 
I have deployed the parser function of bitsight findings in our environment and it would deployed as success without having any error. 

<img width="956" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/073652a2-f361-41bc-b9c9-d6be2e89eb6d">

<img width="281" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/1fc20125-2782-402e-9d51-fefbd099eadd">

Thanks

v-prasadboke commented:
Hello @devikamehra can you please review and approve the analytic rules from the PR

v-prasadboke commented:
Hello @jayeshprajapaticrest Have you recently removed any extra commas ( , ) from the parser, If yes then please repackage the solution.
That is what was causing the parser not to run properly. 

Any changes made to content of the solution should reflect in maintemplate as well.
Which takes place after when you make changes and repackage the solution.

And the maintemplate & createui inside the zip package and in the package folder should be identical.
Thanks Prasad

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest Have you recently removed any extra commas ( , ) from the parser, If yes then please repackage the solution. That is what was causing the parser not to run properly.
> 
> Any changes made to content of the solution should reflect in maintemplate as well. Which takes place after when you make changes and repackage the solution.
> 
> And the maintemplate & createui inside the zip package and in the package folder should be identical. Thanks Prasad

@v-prasadboke 
We checked the parser and fixed the issue related with it.
Can you please again deploy and confirm that its working or not?
Thanks.

Let me know if any other changes required.

v-prasadboke commented:
Ok @jayeshprajapaticrest looking in it

jayeshprajapaticrest commented:
> Ok @jayeshprajapaticrest looking in it

@v-prasadboke Is there any update?

v-prasadboke commented:
Hello @jayeshprajapaticrest facing issue with parsers

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest facing issue with parsers

Can we take some time this week to connect on MS team or Zoom and resolve it as soon as possible to expediate the process?

jayeshprajapaticrest commented on Solutions/BitSight/Parsers/BitSightCompanyDetails.txt:
```diff
@@ -0,0 +1,113 @@
+    | extend
+                EventVendor ="CompanyDetailsBitSight",
+                EventProduct ="CompanyDetailsBitSight",
+                PrimaryCompanyGUID = column_ifexists('primary_company_guid_g', ''),
```
@v-prasadboke 
Understood. But As I see in the sample data the the field name is with the _g suffix(**primary_company_guid_g**) and yes we haven't manually provided the type of this field MS Sentinel automatically append this suffix as the field contains the value as GUID. We are able to update it to primary_company_guid_s based on your suggestion but wherever this field was used is with the _g suffix. And suppose if we update it in the sample data then there might be chance to get an error.

So can you please guide us for the same?

<img width="787" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/cd33c01a-4784-418f-9413-c5bd49041c42">

v-prasadboke commented:
> > Hello @jayeshprajapaticrest facing issue with parsers
> 
> Can we take some time this week to connect on MS team or Zoom and resolve it as soon as possible to expediate the process?

sure it would be very helpfull for the proress, can you share your email id. 
Thank you.

jayeshprajapaticrest commented:
> > > Hello @jayeshprajapaticrest facing issue with parsers
> > 
> > 
> > Can we take some time this week to connect on MS team or Zoom and resolve it as soon as possible to expediate the process?
> 
> sure it would be very helpful for the Solution, can you share your email id so that i can setup a meet. Mine's - [v-prasadboke@microsoft.com](mailto:v-prasadboke@microsoft.com) Thank you.

My MailID: jayesh.prajapati@crestdatasys.com
Can you please share an invite for the same?

v-prasadboke commented:
Hello @jayeshprajapaticrest any reason for closing this PR.

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest any reason for closing this PR.
@v-prasadboke 
I just trying to resolve the conflicts but don't know why it gets automatically closed.
Wait I will do some changes in this PR and then reopen this.

v-prasadboke commented:
Hello @devikamehra can you please review and approve for the analytic rules

jayeshprajapaticrest commented:
@devikamehra @v-prasadboke 
Is there any update in this PR for merge?

v-prasadboke commented:
Hello @devikamehra waiting for your approval

v-prasadboke commented:
This changes were asked earlier but were not committed, I request you to work on these changes.

v-prasadboke commented:
Hello @jayeshprajapaticrest please respond to comments of Devikamehra

v-prasadboke commented:
Hello @jayeshprajapaticrest please respond to the comments of Devika

jayeshprajapaticrest commented:
@v-prasadboke @devikamehra 
I have done with the changes of Entity Mapping. Please confirm and proceed with the merging.
Thanks

v-prasadboke commented:
Hello @devikamehra please review and confirm once

devikamehra reviewed (dismissed):
Analytical rules are validated

v-prasadboke commented:
Hello @devikamehra a validation check is failing for analytic rule, can you please check

v-prasadboke commented:
Hello @jayeshprajapaticrest please make sure offer id and solution id of Solution, should be in lowercase everywhere in the solution

v-prasadboke commented on Solutions/BitSight/Analytic Rules/BitSightDropInCompanyRatings.yaml:
```diff
@@ -0,0 +1,37 @@
+    alertDisplayNameFormat: 'BitSight : Alert for >10% drop in ratings of {{CompanyName}}.'
+    alertDescriptionFormat: 'Alert is generated for {{CompanyName}}.\n\nRating Date: {{RatingDate}}\nPercentage Drop: {{percentage}}%'
+entityMappings:
+  - entityType: Process
```
Hello @jayeshprajapaticrest you have added process entity which is mismatching with identifier. 
please resolve the issue for BitSightDropInHeadlineRating & BitSightDropInCompanyRatings

v-prasadboke commented:
Hello @jayeshprajapaticrest please add template id of all the analytic rules to this skipvalidation file
.script/tests/KqlvalidationsTests/SkipValidationsTemplates.json

v-prasadboke commented:
And as per latest guidelines issued by Microsoft Sentinel, please repackage the solution using V3 tool 
Please go through this readme file on how to use the V3 tool

v-prasadboke commented:
Hello @jayeshprajapaticrest you can continue with current package version

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest you can continue with current package version
@v-prasadboke 
Means with V2 tool right?
Working on to fix the validation error of Analytics rule.

v-prasadboke commented:
Yes, Thank you @jayeshprajapaticrest

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest please add template id of all the analytic rules to this skipvalidation file .script/tests/KqlvalidationsTests/SkipValidationsTemplates.json

@v-prasadboke 
Done as suggested.

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest please make sure offer id and solution id of Solution, should be in lowercase everywhere in the solution

@v-prasadboke 
Done as suggested.

jayeshprajapaticrest commented:
@v-prasadboke Can you proceed with the merging this PR process as early as possible?

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest please add template id of all the analytic rules to this skipvalidation file .script/tests/KqlvalidationsTests/SkipValidationsTemplates.json

@v-prasadboke We did as above still validation error for Analytics rule exist. Can you please help us out with this?

v-prasadboke commented:
Sure @jayeshprajapaticrest, I have scheduled a call for 2:30 PM (IST) for 28 Jul 2023.
You Would have received an invite for the same

v-prasadboke commented:
Hello @jayeshprajapaticrest thank you for committing the changes, but still validation check is failing for this PR.
Will be examining and updating you shortly before 20 Jul 2023

v-prasadboke commented on .script/tests/KqlvalidationsTests/CustomTables/Findings_data_CL.json:
```diff
@@ -0,0 +1,125 @@
+{
+    "Name": "BitSightFindingsData",
```
Hello @jayeshprajapaticrest the value of entity " name is duplicated with customtable BitSightFindingsData, and this table share the same content with BitSightFindingsData.

Please replace the value of entity " name " with any other unique value

jayeshprajapaticrest replied:
@v-prasadboke 
Corrected.

v-prasadboke commented on .script/tests/KqlvalidationsTests/CustomTables/Graph_data_CL.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "Name": "BitSightGraphData",
```
same is the case here Duplicated Value, please replace it with unique value

v-prasadboke commented on .script/tests/KqlvalidationsTests/CustomTables/Breaches_data_CL.json:
```diff
@@ -0,0 +1,69 @@
+{
+    "Name": "BitSightBreaches",
```
Here as well

jayeshprajapaticrest commented:
@v-prasadboke 
I am done with all the changes and there are no validation errors, So can you please proceed with merging of PR?

jayeshprajapaticrest commented:
@v-prasadboke Any update on this PR merging?

v-prasadboke commented:
Hello @jayeshprajapaticrest, Thank you for the commits.
Will be examining this PR and will update you before 26 Jul 2023

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest, Thank you for the commits. Will be examining this PR and will update you before 26 Jul 2023

@v-prasadboke Is it possible to take it on priority. As this PR has almost around 5 month.

v-prasadboke commented:
Hello @jayeshprajapaticrest sorry for the delay in PR review, Will take this on priority and will update you ASAP.

v-prasadboke commented:
Hello @jayeshprajapaticrest are you available

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest are you available

@v-prasadboke Yes

v-prasadboke commented:
Will send you an invitation link for a call, can you join

v-prasadboke commented:
https://teams.microsoft.com/l/meetup-join/19%3ameeting_MjIwOGQyZWEtNjlhZC00NjQxLTk5Y2YtODc3MmU3Zjc4NmU3%40thread.v2/0?context=%7b%22Tid%22%3a%2272f988bf-86f1-41af-91ab-2d7cd011db47%22%2c%22Oid%22%3a%2228263f0a-981b-4a56-a4c9-1e91e9ea44fb%22%7d

jayeshprajapaticrest commented:
@v-prasadboke Joined to above link.

v-prasadboke commented:
Hello @jayeshprajapaticrest please can you confirm on the changes

jayeshprajapaticrest commented:
> Hello @jayeshprajapaticrest please can you confirm on the changes

There is one discrepancy found in your comment. You forgot reverted solution id of Proofpoint after testing.
![image](https://github.com/Azure/Azure-Sentinel/assets/98145046/ebf68d8d-4886-4e94-a4b3-aa1150792b26)

Can you please correct it? Rest looks OK.
Thanks

v-prasadboke commented:
Sorry @jayeshprajapaticrest, and thanks for pointing it out.

jayeshprajapaticrest commented:
> Sorry @jayeshprajapaticrest, and thanks for pointing it out.

@v-prasadboke Thanks for your support and finally approving and merging this PR after long time.
